packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,183 @@
+# Tag 🏷️
+
+These are set of interactive keywords that help organise & categorise objects. Tags can be added or removed from an object by the users.
+
+<img width="426" alt="image" src="https://github.com/razorpay/blade/assets/30949385/75d8068d-d6fc-47e7-aa75-21686ed1c27d">
+
+## Design
+
+- [Tags Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=29210-567576&t=HLRjz3OTofBtFdDl-0)
+
+## Proposed API
+
+```jsx
+<Tag
+  size="medium"
+  onDismiss={() => {
+    console.log('Close Icon Clicked');
+  }}
+>
+  Unpaid
+</Tag>
+```
+
+### Props
+
+```ts
+type TagProps = {
+  /**
+   * Decides the size of Tag
+   *
+   * @default medium
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Callback when close icon on Tag is clicked
+   */
+  onDismiss?: () => void;
+
+  /**
+   * Text that renders inside Tag
+   */
+  children: ReactText;
+};
+```
+
+_Didn't consider any alternate API. Kept `size` and `children` similar to `<Badge />` and `onDismiss` similar to `<Alert />` and `<Dropdown />`_
+
+## Accessibility
+
+- Ensure Close Icon is focussable
+- Close Icon to have appropriate `aria-label`
+
+## Integration with Inputs
+
+<img width="358" alt="image" src="https://github.com/razorpay/blade/assets/30949385/71cd236a-93e7-46f0-9da6-12d519b958d4">
+
+Tags can be added inside other input components such as -
+
+- TextArea
+- SelectInput (in Dropdown)
+- TextInput
+- AutoComplete
+
+<details>
+<summary>Implementation Details</summary>
+
+This will require some refactor in BaseInput to add a slot before the actual Input element.
+
+E.g. This is AutoComplete from Primer where the tags go into a slot and input gets pushed forward. We will be implementing something similar in our BaseInput.
+
+<img width="419" alt="image" src="https://github.com/razorpay/blade/assets/30949385/be01cdfb-1171-4970-b381-76e4c44d1854">
+
+</details>
+
+### Integration with SelectInput and AutoComplete
+
+SelectInput and AutoComplete are not expected to have any changes in consumer API as rendering of tags will be handled internally based on the selected value.
+
+### Integration with TextInput and TextArea
+
+> **Warning**
+>
+> Open Question: Do we want to discuss this right now? or do I remove it from API decision itself?
```

Comment:
So the thing is implementation detail is relevant here. This way of implementation is what becomes a con in alternate API I had proposed. Because the div that has tags is separate from actual input, we can't use `value` prop for adding tags as well.

https://github.com/razorpay/blade/blob/8e47a3da1fc05263aaafa9cd97978133e10bfdf7/packages/blade/src/components/Tag/_decisions/decisions.md?plain=1#L168

Reply:
I have anyway added it inside `<details>` tag so it won't take up too much space while reading API spec